modin/core/dataframe/pandas/metadata/dtypes.py

```diff
@@ -12,13 +12,518 @@
+        # try to compute '._columns_order' using 'new_parent'
+        self.columns_order
+
+    @property
+    def columns_order(self) -> Optional[dict[int, IndexLabel]]:
+        """
+        Get order of columns for the described dataframe if available.
+
+        Returns
+        -------
+        dict[int, IndexLabel] or None
+        """
+        if self._columns_order is not None:
+            return self._columns_order
+        if self._parent_df is None or not self._parent_df.has_materialized_columns:
+            return None
+
+        self._columns_order = {i: col for i, col in enumerate(self._parent_df.columns)}
+        # we got information about new columns and thus can potentially
+        # extend our knowledge about missing dtypes
+        if len(self._columns_order) > (
+            len(self._known_dtypes) + len(self._cols_with_unknown_dtypes)
+        ):
+            new_cols = [
+                col
+                for col in self._columns_order.values()
+                if col not in self._known_dtypes
+                and col not in self._cols_with_unknown_dtypes
+            ]
+            if self._remaining_dtype is not None:
+                self._known_dtypes.update(
+                    {col: self._remaining_dtype for col in new_cols}
+                )
+                self._remaining_dtype = None
+                if len(self._cols_with_unknown_dtypes) == 0:
+                    self._schema_is_known = True
+            else:
+                self._cols_with_unknown_dtypes.extend(new_cols)
+        self._know_all_names = True
+        return self._columns_order
+
+    def __repr__(self):  # noqa: GL08
+        return (
+            f"DtypesDescriptor:\n\tknown dtypes: {self._known_dtypes};\n\t"
+            + f"remaining dtype: {self._remaining_dtype};\n\t"
+            + f"cols with unknown dtypes: {self._cols_with_unknown_dtypes};\n\t"
+            + f"schema is known: {self._schema_is_known};\n\t"
+            + f"has parent df: {self._parent_df is not None};\n\t"
+            + f"columns order: {self._columns_order};\n\t"
+            + f"know all names: {self._know_all_names}"
+        )
+
+    def __str__(self):  # noqa: GL08
+        return self.__repr__()
+
+    def lazy_get(
+        self, ids: list[Union[IndexLabel, int]], numeric_index: bool = False
+    ) -> "DtypesDescriptor":
+        """
+        Get dtypes descriptor for a subset of columns without triggering any computations.
+
+        Parameters
+        ----------
+        ids : list of index labels or positional indexers
+            Columns for the subset.
+        numeric_index : bool, default: False
+            Whether `ids` are positional indixes or column labels to take.
+
+        Returns
+        -------
+        DtypesDescriptor
+            Descriptor that describes dtypes for columns specified in `ids`.
+        """
+        if len(np.unique(ids)) != len(ids):
```

Comment:
In all other places, you seem to use `set` to check for the uniqueness of columns. Why is it different here?

Reply:
there's no reason, will change this to use the same thing everywhere for consistency

Reply:
changed to `np.unique` everywhere

Reply:
changed to using `set` everywhere instead, as numpy seems to not able to process multi-level columns:
```python
>>> np.unique([("level1", "level2"), "col2", ("level1", "level2"), "col3"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 200, in unique
  File "C:\Users\dchigare\Miniconda3\envs\modin\lib\site-packages\numpy\lib\arraysetops.py", line 272, in unique
    ar = np.asanyarray(ar)
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (4,) + inhomogeneous part.
>>> set([("level1", "level2"), "col2", ("level1", "level2"), "col3"])
{'col3', ('level1', 'level2'), 'col2'}
```